daemon/common.h

```diff
@@ -26,6 +26,11 @@
 
 #define config_generate(buffer, only_changed) appconfig_generate(&netdata_config, buffer, only_changed)
 
+// ----------------------------------------------------------------------------
+// shortcuts for common patterns
+#define yield() if(unlikely(netdata_exit)) break
```

Comment:
I think this will introduce a nasty issue.

Example:

```c
if(x == 0)
  yeld();
else
  something_else();
```

When will `something_else()` run?

I know this is not used like that today, but then all contributors will have to remember that `yield()` is a special case, and cannot be used like all other functions.

Reply:
The generally acceptable way for doing things like this, is:

```c
#define yield() do { if(unlikely(netdata_exit)) break; } while(0)
```

The above does not have the problem, but `break`, breaks that `while()` loop, not the indented one.

Another way would be to define it like that:

```c
#define IF_WE_HAVE_TO_EXIT_BREAK_THE_LOOP if(unlikely(netdata_exit)) break
```

This would be better, because people would understand this is a special macro.
Generally however, I don't think it will add much value compared to the existing.

Reply:
> When will `something_else()` run?

You're right. This can be amended by making this macro a full `if`-`else` block, considering that the `else` belongs to the closest `if`.

Reply:
> I think this will introduce a nasty issue.
> ...
> When will `something_else()` run?

This is one important reason to *not* skip curly braces for one-liners. Additionally, the braces make the code much easier to understand.